scripts/generate-app-config/generatePagesConfig.js

```diff
@@ -144,7 +144,10 @@ const generatePagesConfig = (siteConfig, production, verbose) => {
     root, source, dist, entryPoint,
   }) => {
     const rootPath = root.replace(/[\\]/g, '/');
-    const sourceDir = ((!production && hotReloading) ? `${source}/` : `${dist}/`) || '';
+    let sourceDir = '';
+    if (source && dist) {
```

Comment:
Is this correct when source OR dist are provided?

Reply:
Maybe this?

```javascript
let sourceDir = ((!production && hotReloading) ? source :  dist) || '';
sourceDir = sourceDir ? `${sourceDir}/` : sourceDir;
```

Reply:
So the scenerio I am trying to account for is:
```
 generatePages: {
    searchPatterns: [
      {
        root: process.cwd(),
        entryPoint: 'terra-dev-site',
      },
    ],
  },

```

Reply:
So that would be option 1 of the config outlined in the [documentation](https://github.com/cerner/terra-dev-site/blob/master/src/terra-dev-site/doc/Configuration.b/site.config.a.doc.md)
```
generatePages: {
    searchPatterns: [
      // In Prod & Dev & !hotReloading: Searches rootPaths/terra-dev-site/**/*{pageTypes}.{jsx,js,md,}
      // In Prod & Dev & !hotReloading: Searches rootPaths/packages/*/terra-dev-site/**/*{pageTypes}.{jsx,js,md,}
      {
        root: process.cwd(),
        entryPoint: 'terra-dev-site',
      },
      // In Prod & Dev & !hotReloading: Searches rootPaths/dist/examples/**/*{pageTypes}.{jsx,js,md,}
      // In Prod & Dev & !hotReloading: Searches rootPaths/packages/*/dist/examples/**/*{pageTypes}.{jsx,js,md,}
      {
        root: process.cwd(),
        dist: 'dist',
        entryPoint: 'examples',
      },
      // In Prod & !hotReloading: Searches rootPaths/lib/terra-dev-site/**/*{pageTypes}.{jsx,js,md,}
      // In Prod & !hotReloading: Searches rootPaths/packages/*/lib/terra-dev-site/**/*{pageTypes}.{jsx,js,md,}
      // In Dev & hotReloading: Searches rootPaths/src/terra-dev-site/**/*{pageTypes}.{jsx,js,md,}
      // In Dev & hotReloading: Searches rootPaths/packages/*/src/terra-dev-site/**/*{pageTypes}.{jsx,js,md,}
      {
        root: process.cwd(),
        source: 'src',
        dist: 'lib',
        entryPoint: 'terra-dev-site',
      },
```

We by default provide source as `src` and dist as `lib` but I am overriding this in my config